### PR TITLE
Create vendor folder and fresh dist as a target only without extras including mathjax

### PR DESCRIPTION
**Pull request by archmoj**
Following the previous conversation on slack,
it seems reasonable (and less confusing) to move `extras/mathjax` and `topojson` to `vendor` folder and start a fresh `dist`.

TODOs: 
- [x] figure out if all/any of the dist files we want to keep in the `dist` e.g. `dist/README.md`
- [x] rewrite the `requirejs` added in #914 so that it does not depend on the old build in dist and really tests what it should test

Next steps:
- Remove `mathjax` from `vendor` and create a script to download, complete and update `MathJax-v2` files from v2.3 to 2.7.x

@plotly/plotly_js

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -132,6 +133,7 @@
     "browserify": "^17.0.0",
     "browserify-transform-tools": "^1.7.0",
     "bubleify": "^2.0.0",
+    "canvas": "^2.6.1",
```
Comment (archmoj) [inline package.json:136]: https://www.npmjs.com/package/canvas

File: devtools/test_dashboard/index.html [hunk 2]
```diff
@@ -19,7 +19,7 @@
   </div>
   <div id="snapshot"></div>
 
-  <script src="../../dist/extras/mathjax/MathJax.js?config=TeX-AMS-MML_SVG"></script>
+  <script src="../../vendor/extras/mathjax/MathJax.js?config=TeX-AMS-MML_SVG"></script>
```
Comment (alexcjohnson) [inline devtools/test_dashboard/index.html:22]: Can we drop the `extras` here? Just `vendor/mathjax`?

Comment (archmoj) [reply devtools/test_dashboard/index.html:22]: We can. But we are going to remove this folder right after to upgrade to 2.5.1 then 2.7.x.
Do you still prefer moving those?

**Review by alexcjohnson — APPROVED** [review]
💃 Looks great - thanks for bringing back (and fixing) the requirejs test!

**Pull request merged**